Source/Generators/LMCArraySignalGenerator.cc

```diff
@@ -409,28 +417,33 @@ namespace locust
 
         allRxChannels.resize(nChannels);
 
-        for(int channelIndex = 0; channelIndex < nChannels; ++channelIndex)
+        for (int subarrayIndex = 0; subarrayIndex < nSubarrays; ++subarrayIndex)

```

Comment:
To me it makes sense to do:
for each channel:
  for each subarray:
    for each receiver: 
That way, the theta can still be calculated accurately. 
On a side note, we should also think about how this affects the reconstruction with Katydid.